js/index.js

```diff
@@ -1,16 +1,47 @@
-import device;
-import lib.PubSub;
-import lib.Callback;
+var FacebookInstant = Class(function () {
+  "use strict";
 
+  this.FBInstant = window.FBInstant;
 
-exports.onReady = new lib.Callback();
+  this.initialise = function (cb, opts) {
+    var FBInstant = this.FBInstant;
 
-init();
+    FBInstant.initializeAsync()
+      .then(function () {
+        FBInstant.setLoadingProgress(100);
+        FBInstant.startGameAsync()
+        .then(function (){
+          cb(opts);
+        })
+      });
+  };
 
-function init() {
-  var FB = window.FBInstant;
-  FBInstant.initializeAsync()
-    .then(function() {
-      exports.onReady.fire();
-    });
-}
+  this.getUserInfo = function () {
+    "use strict";
+
+    var FBInstant = window.FBInstant,
+      contextId = FBInstant.context.getID(),
+      contextType = FBInstant.context.getType(),
+      playerName = FBInstant.player.getName(),
+      playerPic = FBInstant.player.getPhoto(),
+      playerId = FBInstant.player.getID();
+
+      return {
+        name: playerName,
+        id: playerId,
+        user_pic: playerPic,
+        cxt_type: contextType,
+        cxt_id: contextId
+      };
+  };
+
+  this.getUserPhoto = function () {
```

Comment:
let's try to mimick FBInstant API as much as possible.

Reply:
yup